src/Microsoft.AspNetCore.Http.Connections.Client/HttpConnection.cs

```diff
@@ -229,6 +233,8 @@ internal HttpConnection(HttpConnectionOptions httpConnectionOptions, ILoggerFact
 
                 foreach (var transport in negotiationResponse.AvailableTransports)
                 {
+                    cancellationToken.ThrowIfCancellationRequested();
```

Comment:
This is probably another one of my biggest beef with CT. You just sprinkle calls to this in arbitrary places in the code. You could also check cancellationToken.IsCancellationRequested to break out of the loop early...

Reply:
The question is how does the user want to react to cancellation. I don't like that it uses exceptions but that seems to be the common pattern. If we just break the loop, you'd get the error that "no transport could be connected" rather than an `OperationCancelledException`